app/src/main/java/com/resala/mobile/qrregister/ui/eventsFragment/EventsAdapter.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Created by  Mobile Dev Team  on 2/3/20 11:33 AM
+ * Copyright (c) Resala Charity Organization. All rights reserved.
+ */
+
+package com.resala.mobile.qrregister.ui.eventsFragment
+
+import android.renderscript.ScriptGroup
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.resala.mobile.qrregister.R
+import com.resala.mobile.qrregister.databinding.ViewEventListItemBinding
+import com.resala.mobile.qrregister.shared.data.model.EventPOJO
+import com.resala.mobile.qrregister.shared.interfaces.SimpleItemClickListener
+import kotlinx.android.synthetic.main.dialog_scan_success.view.*
+import java.text.FieldPosition
+
+class EventsAdapter<T>(private val eventList: ArrayList<T>) : RecyclerView.Adapter<EventsAdapter.EventListViewHolder>() {
+
+    private lateinit var mClickListener:SimpleItemClickListener
```

Comment:
try instead of click  listener interface to use kotlin awesome feature  higher order function

Reply:
I updated it, please review it and tell if there is anything wrong